shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm

```diff
@@ -6,13 +6,11 @@
 
 #import "flutter/shell/platform/darwin/ios/framework/Source/SemanticsObject.h"
 
-@interface FlutterSemanticsScrollView ()
-@property(nonatomic, assign) SemanticsObject* semanticsObject;
-@end
-
-@implementation FlutterSemanticsScrollView
+@implementation FlutterSemanticsScrollView {
+  fml::WeakPtr<SemanticsObject> _semanticsObject;
+}
```

Comment:
Is there something special about `WeakPtr`, or does it work if you keep the Objective-C property and just make it weak?
```suggestion
@interface FlutterSemanticsScrollView ()
@property(nonatomic, weak) SemanticsObject* semanticsObject;
@end
```

Reply:
You can't use weak in MRC right?

Reply:
Argh right, it's already `assign`.  And I can't remember how to get this pattern to not `message sent to deallocated instance`.

Reply:
Oh, I remember, `FlutterScrollableSemanticsObject` would have to set the scrollView's semanticsObject to `nil` in dealloc.

Reply:
Yea, you only get auto nilling variables with ARC.

Who is the real owner of the semanticsObject?  Do they have a reference to the FlutterSemanticsScrollView?  They can nil out the property.

Reply:
> Oh, I remember, `FlutterScrollableSemanticsObject` would have to set the scrollView's semanticsObject to `nil` in dealloc.

https://github.com/flutter/engine/blob/48ce284dff9cd876486542b0c8df850c9ffd5903/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L168-L170
like
https://github.com/flutter/engine/blob/48ce284dff9cd876486542b0c8df850c9ffd5903/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L301

Reply:
I tried something close to this and couldn't get it working.  Should I try again or just keep with the weak factory?